### DISCUSSION TITLE
Disabled input in Safari do not show text color correctly

### DISCUSSION THREAD

**Discussion by fabioborella-smash** [tailwindlabs/tailwindcss#6294] [Help]
Hi everyone,

is there a way to enable 'currentcolor' to work on Safari and display correctly the text-* color classes on disabled inputs? 
It seems to happen only in Safari, and the only fix I found is to manually style the input with -webkit-text-fill-color, but it is not very framework oriented.

Tailwind Play link to show bug:
https://play.tailwindcss.com/ScVIq5xdZJ

Thanks in advance!

**Comment by Fruchtix** [comment]
For me it worked to use the readonly input attribute instead of disabling the input.

**Reply by majohannsen** [reply]
That's a nice workaround but with readOnly you can still tab through your inputs, with disabled you can't.

**Comment by Merott** [comment]
This seems like a limitation of Safari that doesn't properly handle `-webkit-text-fill-color` being set to `currentColor`.

The simplest (yet not ideal) solution I've found is to set `-webkit-text-fill-color` as an arbitrary property:

```html
<input class="[-webkit-text-fill-color:theme(colors.yellow.500)]" />
```

Another solution (still not ideal) would be to define a [custom plugin](https://v1.tailwindcss.com/docs/plugins#adding-base-styles) to globally set both `-webkit-text-fill-color` and the `color` properties as base styles on disabled inputs:

```js
plugin(function ({ addBase, config }) {
  const selectors = [
    'input:disabled',
    'textarea:disabled',
    'input:disabled::placeholder',
    'textarea:disabled::placeholder',
  ]

  addBase({
    ':root': {
      '--disabled-input-color': config('theme.colors.gray.400'),
      // to override: [--disabled-input-color:theme(colors.yellow.500)]
    },
    [selectors.join(', ')]: {
      '-webkit-text-fill-color': 'var(--disabled-input-color)',
      'color': 'var(--disabled-input-color)',
    },
  })
})
```

You can then override that color through `--disabled-input-color`, i.e.

```html
<input class="[--disabled-input-color:theme(colors.yellow.500)]" />
```

**Comment by basememara** [comment]
Encountered this issue as well and @Merott's plugin idea is great but I wanted to zero in on Safari and that seemed to get applied to more than Safari. I also opted for a custom variant:

```js
plugin(({ addVariant }) => addVariant('safari', '@supports (-webkit-hyphens:none)'))
```

Then I use it like this:

```js
<input class="text-gray-500 safari:text-gray-700" />
```

Btw is there an out-of-the-box approach to only target Safari in Tailwind? I used an at-rule but who knows when a Safari update will break it. Reference: https://stackoverflow.com/a/25975282
